src/DistributedComputations/distributed_architectures.jl

```diff
@@ -51,10 +23,46 @@ Return `Partition` representing the division of a domain into
 `Rx` parts in `x` and `Ry` parts in `y` and `Rz` parts in `z`,
 where `x, y, z` are the first, second, and third dimension
 respectively.
+
+`Rx`, `Ry` and `Rz` can be vectors containing the percentage of the domain
+ascribed to the different cores. In this case, `length(Rx)` represent the number
+of divisions in the first dimension with `sum(Rx)` equal to 1.0. 
+Rank `arch.local_index[i]` will have `global_size[i] * Rx[i]` elements in the first dimension.
 """
-Partition(Rx::Number, Ry::Number=1, Rz::Number=1) = Partition{Rx, Ry, Rz}()
-Partition(lengths::Array{Int, 1}) = Partition(reshape(lengths, length(lengths), 1))
-Base.size(::Partition{<:Any, Rx, Ry, Rz}) where {Rx, Ry, Rz} = (Rx, Ry, Rz)
+Partition(Rx)     = Partition(validate_partition(Rx, 1, 1)...)
+Partition(Rx, Ry) = Partition(validate_partition(Rx, Ry, 1)...)
+
+function Partition(; x = 1, y = 1, z = 1)
+    x, y, z = validate_partition(x, y, z)
+    return Partition(x, y, z)
+end
+
+struct Equal end
+struct Fractional{S} end
+struct Sizes{S} end
+
+ranks(p::Partition) = (ranks(p.Rx), ranks(p.Ry), ranks(p.Rz))
+ranks(r::Int)      = r
+ranks(r::Sizes)    = length(r)
+ranks(r::Fractional) = length(r)
+
+Base.size(p::Partition) = ranks(p)
+
+Fractional(args...) = sum(args) != 1 ? Fractional{args ./ sum(args)}() : Fractional{args}()
```

Comment:
Why the if statement? If `sum(args) == 1` then `args ./ sum(args) == args` right?

Reply:
to normalize the arguments to fractional values. Maybe we can issue a warning before so that the user knows we are normalizing the partitions?

Reply:
This code does the same thing:

```julia
Fractional(args...) = Fractional{args ./ sum(args)}()
```

there is no need for an `if`-statement. It doesn't change anything.

Reply:
oh yeah right